release-20.1: backupccl: all full cluster restore of backup with no user data

pbardea opened the pull request:
Backport 1/1 commits from #49591.

/cc @cockroachdb/release

---

This commit fixes a bug in which full cluster restore assumed that a
full cluster backup had at least one user database/table to restore.

Fixes #49573.

Release note (bug fix): Previously, if one attempted to perform a full
cluster RESTORE on a backup that did not contain any user data it would
fail.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/49745)
<!-- Reviewable:end -->

pbardea committed:
backupccl: all full cluster restore of backup with no user data

This commit fixes a bug in which full cluster restore assumed that a
full cluster backup had at least one user database/table to restore.

Release note (bug fix): Previously, if one attempted to perform a full
cluster RESTORE on a backup that did not contain any user data it would
fail.
